device/celestica/x86_64-cel_questone2bd-r0/sonic_platform/fan.py

```diff
@@ -0,0 +1,295 @@
+#!/usr/bin/env python
+
+#############################################################################
+# Celestica
+#
+# Module contains an implementation of SONiC Platform Base API and
+# provides the fan status which are available in the platform
+#
+#############################################################################
+
+import json
+import math
+import os.path
+
+try:
+    from sonic_platform_base.fan_base import FanBase
+    from helper import APIHelper
+except ImportError as e:
+    raise ImportError(str(e) + "- required module not found")
+
+
+FAN_NAME_LIST = ["FAN-1F", "FAN-1R", "FAN-2F",
+                 "FAN-2R", "FAN-3F", "FAN-3R", "FAN-4F", "FAN-4R"]
+
+FAN_SYSFS_PATH = "/sys/bus/i2c/drivers/fancpld/66-000d/"
+FAN_DIRECTION_BIT = 1
+FAN_PRESENT_BIT = 1
+FAN_RPM_MULTIPLIER = 150
+FAN_MAX_PWM = 255
+FAN_MAX_RPM = FAN_MAX_PWM * FAN_RPM_MULTIPLIER
+FAN_PRESENT_SYSFS = "fan{}_present"
+FAN_PWM_SYSFS = "fan{}_pwm"
+FAN_DIRECTION_SYSFS = "fan{}_dir"
+FAN_LED_SYSFS = "fan{}_led"
+FAN_LED_GREEN_CMD = "0x1"
+FAN_LED_RED_CMD = "0x2"
+FAN_LED_OFF_CMD = "0x3"
+
+PSU_FAN_MAX_RPM = 30000
+PSU_HWMON_PATH = "/sys/bus/i2c/devices/i2c-{0}/{0}-00{1}/hwmon"
+PSU_I2C_MAPPING = {
+    0: {
+        "i2c_num": 75,
+        "pmbus_reg": "58",
+    },
+    1: {
+        "i2c_num": 76,
+        "pmbus_reg": "59",
+    },
+}
```

Comment:
Please check these PSU indexes same as in psu.py

Reply:
fixed https://github.com/SONIC-DEV/sonic-buildimage/pull/149/commits/f5c887881d241c6831c82e738d765fc4d71b6a6e